File: app/src/main/java/com/example/pam_project/lists/dialogs/ListActivityDialogFragment.java
package com.example.pam_project.lists.dialogs;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

public abstract class ListActivityDialogFragment extends DialogFragment {
    protected SelectedDialogItems callback;

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        try {
            callback = (SelectedDialogItems) getActivity();
        } catch (ClassCastException e) {
            Log.d(this.getClass().getName(), "Activity doesn't implement the SelectedDialogItems interface");
        }
    }

    @Override
    public void onDetach() {
        callback = null;
        super.onDetach();
    }
}

File: app/src/main/java/com/example/pam_project/dialogs/ListActivityDialogFragment.kt
package com.example.pam_project.dialogs

import android.content.Context
import android.util.Log
import androidx.fragment.app.DialogFragment

abstract class ListActivityDialogFragment : DialogFragment() {
    protected var callback: SelectedDialogItems? = null
    override fun onAttach(context: Context) {
        super.onAttach(context)
        try {
            callback = activity as SelectedDialogItems?
        } catch (e: ClassCastException) {
            Log.d(this.javaClass.name, "Activity doesn't implement the SelectedDialogItems interface")
        }
    }

    override fun onDetach() {
        callback = null
        super.onDetach()
    }
}